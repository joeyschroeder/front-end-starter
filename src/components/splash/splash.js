import React from 'react';
import { Screen } from 'components/screen/screen';
import styles from './splash.style.scss';

export const Splash = () => {
  return (
    <Screen centerContent className={styles.root}>
      <div className={styles.primary}>
        <h1>
          <strong>Front-End</strong> Starter
        </h1>
        <p>
          This project is designed to get you up and running with your next
          front-end, JavaScript application faster than ever! This project
          includes <strong>React</strong>, <strong>Redux</strong>, and many
          other utilities for easily maintaining a front-end application.
        </p>
      </div>
    </Screen>
  );
};
