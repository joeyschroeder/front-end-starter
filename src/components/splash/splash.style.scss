@import "../../styles/config/variables";
@import "../../styles/mixins/font-size";
@import "../../styles/mixins/line-height";

.root {
  h1 {
    @include font-size(5);
    @include line-height(3.8);

    font-weight: $font-weight-light;
    text-transform: uppercase;

    strong {
      font-weight: $font-weight-heavy;
    }

    @media screen and (max-width: $width-mobile) {
      @include font-size(4);
      @include line-height(3);
    }
  }
}

.primary {
  max-width: $width-mobile;
}
