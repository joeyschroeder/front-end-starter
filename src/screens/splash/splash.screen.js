import React from 'react';
import { Screen } from '../../components/screen/screen.component';
import styles from './splash.style.scss';

export const SplashScreen = () => {

    return (
        <Screen centerContent className={styles.root}>
            <div className={styles.primary}>
                <h1><strong>Front-End</strong> Starter</h1>
                <p>This project is designed to get you up and running with your next front-end, JavaScript application faster than ever!  This project includes <strong>React</strong>, <strong>Redux</strong>, and many other utilities for easily maintaining a front-end application.</p>
            </div>
        </Screen>
    );
};
