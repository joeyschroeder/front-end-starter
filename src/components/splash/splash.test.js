/* global React, create */
import { Splash } from './splash';

describe('<Splash />', () => {
  it('should render correctly', () => {
    const component = create(<Splash />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
