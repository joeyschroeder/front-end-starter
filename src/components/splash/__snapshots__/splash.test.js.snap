// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Splash /> should render correctly 1`] = `
<div
  className="animated fadeIn root root centerContent"
>
  <div
    className="primary"
  >
    <h1>
      <strong>
        Front-End
      </strong>
       Starter
    </h1>
    <p>
      This project is designed to get you up and running with your next front-end, JavaScript application faster than ever! This project includes 
      <strong>
        React
      </strong>
      , 
      <strong>
        Redux
      </strong>
      , and many other utilities for easily maintaining a front-end application.
    </p>
  </div>
</div>
`;
